torch/csrc/jit/passes/quantization.cpp

```diff
@@ -1306,22 +1304,29 @@ void InsertQuantDeQuantHelper::quantizeTensors(
     Module& module,
     Graph* g,
     Value* self) {
-  if (!observer_nodes_.count(g)) {
+  if (!observer_nodes_for_graph_.count(g)) {
     return;
   }
-  for (auto* n : observer_nodes_.at(g)) {
+  for (auto* n : observer_nodes_for_graph_.at(g)) {
     auto* original_value = n->input(1);
     auto tp = getQSchemeAndQParamVector(module, n);
     auto qscheme = std::get<0>(tp);
     auto qparam_map = std::get<1>(tp);
     checkQScheme(g, qscheme);
+    std::vector<std::string> qparam_names;
     for (auto& pr : qparam_map) {
       const auto& name = pr.first;
       const auto& qparam = pr.second;
-      module.register_attribute(
-          original_value->debugName() + name, qparam.type(), qparam);
+      size_t uid = 0;
+      auto qparam_name = original_value->debugName() + name + "_" + c10::to_string(uid++);
+      while (module.hasattr(qparam_name)) {
```

Comment:
Why a while loop?

Reply:
we need to make sure the attribute name is unique